Assets/Mirage/Weaver/TypeExtensions.cs

```diff
@@ -22,20 +24,31 @@ public static List<MethodDefinition> GetMethods(this TypeDefinition td, string m
             return td.Methods.Where(method => method.Name == methodName).ToList();
         }
 
-        public static MethodDefinition GetMethodInBaseType(this TypeDefinition td, string methodName)
+        public static MethodReference GetMethodInBaseType(this TypeReference td, string methodName)
         {
-            TypeDefinition typedef = td;
+            TypeDefinition typedef = td.Resolve();
+            TypeReference typeRef = td;
             while (typedef != null)
             {
                 foreach (MethodDefinition md in typedef.Methods)
                 {
                     if (md.Name == methodName)
-                        return md;
+                    {
+                        MethodReference method = md;
+                        if (typeRef.IsGenericInstance)
+                        {
+                            var baseTypeInstance = (GenericInstanceType)typeRef;
+                            method = method.MakeHostInstanceGeneric(baseTypeInstance);
```

Comment:
It is not that simple,  consider this case:
```cs
class Grampu<T> {
      public T Lovely() { ...}
}

class Papu<T> : Grampu<int> {
}

class Oobie : Papu<T> {
}
```

if you try to get the method Lovely() from Oobie,  you should get Grampu<int>.Lovely(),  but with your code, you will get Grampu<T>.Lovely()

Reply:
Fixed in the cfc9a6f.